projects/ffmpeg/build.sh

```diff
@@ -0,0 +1,242 @@
+      export TARGET_BSF="/ffmpeg/tools/target_bsf_fuzzer.c -o /target_bsf_fuzz"
+      export TARGET_DEC="-lxcb -lxcb-shm -lxcb -lxcb-xfixes -lxcb -lxcb-shape -lxcb -lX11 \
+                         -DFFMPEG_CODEC=AV_CODEC_ID_MPEG1VIDEO -DFUZZ_FFMPEG_VIDEO \
+                         /ffmpeg/tools/target_dec_fuzzer.c -o /target_dec_fuzz"
+      export TARGET_DEM="-DIO_FLAT=0 /ffmpeg/tools/target_dem_fuzzer.c -o /target_dem_fuzz"
+
+      mkdir -p $OUT/lib/
+      cp /usr/lib/x86_64-linux-gnu/libbz2.so.1.0 $OUT/lib/
+      cp /usr/lib/x86_64-linux-gnu/libz.so.1 $OUT/lib/
+      bzip2 -f -d alsa-lib-1.1.0.tar.bz2
+fi
+
+if [[ $CONFIG = "afl" ]]
+then
+      export CC="afl-clang-fast"
+      export CXX="afl-clang-fast++"
+      export CFLAGS="-g -fsanitize=address,integer,bounds,null,undefined,float-divide-by-zero"
+      export CXXFLAGS="-g -fsanitize=address,integer,bounds,null,float-divide-by-zero -fno-sanitize=vptr"
+      export LINK_FLAGS="-g -fsanitize=fuzzer,address,integer,bounds,null,undefined,float-divide-by-zero $LINK_LIBS"
+      export TARGET_BSF="/ffmpeg/tools/target_bsf_fuzzer.c -o /target_bsf_afl"
+      export TARGET_DEC="-lxcb -lxcb-shm -lxcb -lxcb-xfixes -lxcb -lxcb-shape -lxcb -lX11 -lbz2 \
+                         -DFFMPEG_CODEC=AV_CODEC_ID_MPEG1VIDEO -DFUZZ_FFMPEG_VIDEO \
+                         /ffmpeg/tools/target_dec_fuzzer.c -o /target_dec_afl"
+      export TARGET_DEM="-DIO_FLAT=0 /ffmpeg/tools/target_dem_fuzzer.c -o /target_dem_afl"
+fi
+
+if [[ $CONFIG = "sydr" ]]
+then
+      export CC=clang
+      export CXX=clang++
+      export CFLAGS="-g"
+      export CXXFLAGS="-g -fno-sanitize=vptr"
+      export LINK_FLAGS="-g $LINK_LIBS"
+      export TARGET_BSF="-o /target_bsf_sydr target_bsf_fuzzer.o main.o"
+      export TARGET_DEC="-lxcb -lxcb-shm -lxcb -lxcb-xfixes -lxcb -lxcb-shape -lxcb -lX11 -lbz2 \
+                         -o /target_dec_sydr target_dec_fuzzer.o main.o"
+      export TARGET_DEM="-o /target_dem_sydr target_dem_fuzzer.o main.o"
+fi
+
+if [[ $CONFIG = "coverage" ]]
+then
+      export CC=clang
+      export CXX=clang++
+      export CFLAGS="-g -fprofile-instr-generate -fcoverage-mapping"
+      export CXXFLAGS="-g -fprofile-instr-generate -fcoverage-mapping -fno-sanitize=vptr"
+      export LINK_FLAGS="$CFLAGS $LINK_LIBS"
+      export TARGET_BSF="-o /target_bsf_cov target_bsf_fuzzer.o main.o"
+      export TARGET_DEC="-lxcb -lxcb-shm -lxcb -lxcb-xfixes -lxcb -lxcb-shape -lxcb -lX11 -lbz2 \
+                         -o /target_dec_cov target_dec_fuzzer.o main.o"
+      export TARGET_DEM="-o /target_dem_cov target_dem_fuzzer.o main.o"
+fi
+
+# Disable UBSan vptr since several targets built with -fno-rtti.
+
+# Build dependencies.
+export FFMPEG_DEPS_PATH=/ffmpeg_deps
+rm -rf $FFMPEG_DEPS_PATH
+mkdir -p $FFMPEG_DEPS_PATH
+
+export PATH="$FFMPEG_DEPS_PATH/bin:$PATH"
+export LD_LIBRARY_PATH="$FFMPEG_DEPS_PATH/lib"
```

Comment:
Do we need this variable? It is used when searching for dynamic libraries at the time of executing the binary, but setting environment variable here won't set it for the container, only for build script execution.

Reply:
we need this variable for coverage build. i've tried to build without it, but there were errors.

Reply:
okay, let it be